### ISSUE TITLE
FvwmPager doesn't update current desk durning initialization

### ISSUE THREAD

**Issue by rasatpc** [fvwmorg/fvwm3#1122]
## Upfront Information
* Fvwm3 version (run: `fvwm3 --version`)
fvwm3 1.1.1 (1.1.0-129) with issue
fvwm3 1.1.1 (1.1.0-119-g360b5bb2) with issue (maybe earlier)

fvwm3 1.1.1 (1.1.0-72) is ok

## Expected Behaviour
Remember the Desk when changing from one FvwmPager to another.

## Actual Behaviour
Working on Desk 2 (`DesktopName 2 Office`) in config "VdeskPager". When changed to config BarButtons with `Swallow "FvwmPagerBar"` it shows Desk 0 (`DesktopName 0 Web`). In WorkPager it is ok.

Screenshot:
https://rasatpc.net/screenshot/fvwm3/pager-change-issue.png

**Comment by somiaj** [comment]
Can you share the configuration and how you launched the problematic pager?

**Comment by rasatpc** [comment]
1. Load config with `VdeskPager`
https://github.com/rasatpc/Fvwm-KISe/blob/main/core/7Modules/WorkVpager/wVPager.sys

2. Click on `Desk 2 (DesktopName 2 Office)`

3. This changes `VdeskPager` to `BarButtons` (`Swallow "FvwmPagerBar"`)
https://github.com/rasatpc/Fvwm-KISe/blob/main/core/7Modules/Desker/dButtonPager.sys

**Comment by somiaj** [comment]
`*BarButtons: (4x2, Swallow "FvwmPagerBar" 'FvwmPager FvwmPagerBar * *')` -- The syntax is incorrect, only use a single `*`, not two.

**Comment by somiaj** [comment]
Well in my testing, I guess two stars, `* *` does work, though only one is needed. I am unable to reproduce this issue.

**Comment by rasatpc** [comment]
It seems the issue is with `GoToDesk`. I did a test with FvwmConsole, randomly selecting `GoToDesk 1 0 0`, ... `3 0 0`, ... `2 0 0`. The results were unpredictable. Only `GoToDesk 0 0 0` is okay.

fvwm3 1.1.1 (1.1.0-130-g27ffd68a)

**Comment by somiaj** [comment]
I still cannot produce the issue with FvwmPager not updating the current desk properly. But the commands you are issuing are mostly nonsense (though it does appear they are broken I'll look into that). But `GotoDesk 3 0 0` is not correct use of the command. From the manual page:

"Three arguments specify a relative desk and the minimum and maximum of the allowable range."

So you are stating move `3` desks from the current desk, but the minimum and maximum desks are both `0`, so in essence this should move you to desk `0`, but there is a bug in the looping code. Probably overlooked since this command makes very little sense to actually use.

**Comment by rasatpc** [comment]
>> But GotoDesk 3 0 0 is not correct use of the command. From the manual page.
https://www.fvwm.org/Wiki/Config/PagesAndDesks/

**Comment by somiaj** [comment]
Maybe read my full post, and notice how no example on the wiki uses the min and max desktop to be the same. Though I fixed the bug that existed due to 'wrapping' not working correctly with more than a single wrap. My point was that command is not one someone should be using, which is most likely why the bug hasn't been noticed until now.

**Comment by somiaj** [comment]
Your version number doesn't make sense, the version I see from main is `1.1.1 (1.1.0-130-g27ffd68a3)`, maybe share the full first line of `fvwm3 --version`. Anyways, I'm unable to reproduce the issue, so until I can reproduce it I cannot debug the issue. You should work on creating a minimal example on top of the default-config that causes the issue.

**Comment by rasatpc** [comment]
>> You should work on creating a minimal example on top of the default-config that causes the issue.

This example config was tested in version 1.1.0-72 and 1.1.0-132  with the default-config. Screenshot result.
https://rasatpc.net/screenshot/fvwm3/test-v110-72-with-v110-132.png

```
## miniPager.sys

DesktopSize 2 1

Module FvwmPager miniPager *

DestroyModuleConfig miniPager: *
*miniPager: Geometry -180+70
*miniPager: Back #d9d9d9
*miniPager: HilightColorset * 6
*miniPager: BalloonColorset * 10
*miniPager: WindowColorsets 10 11
*miniPager: Font "xft:Sans:size=12:antialias=True"
*miniPager: SolidSeparators
*miniPager: WindowBorderWidth 1
*miniPager: BalloonStringFormat %c
*miniPager: BallonYOffset +2
*miniPager: Window3dBorders
*miniPager: UseSkipList
*miniPager: MiniIcons

Style "miniPager" NoTitle, !Handles, Borders, BorderWidth 2, WindowListSkip, Sticky, StaysOnTop
```

**Comment by somiaj** [comment]
Okay it took a while to figure out what the bug actually was (partly because your description is fairly poor). But after looking at the screenshots I finally figured out what you mean. If you run FvwmPager on a desk other than 0, it won't be updated until an update desk change happens. I'll look into that.

**Comment by somiaj** [comment]
Okay, I see what is going on and fixed the issue. Check out [js/gh-1122](https://github.com/fvwmorg/fvwm3/tree/js/gh-1122).

In the future try to better describe what the issue is as trying to fish use information makes it way harder to track down the actual bug. This has nothing to do with changing desks, or remembering what the current desk is. This is about FvwmPager updating the current desk during initialization.

**Comment by rasatpc** [comment]
Thanks for the fix,` fvwm3 1.1.2 (1.1.1-2-gefef84cf)` works fine.
